patches/server/0988-Deprecate-and-replace-methods-with-old-StructureType.patch

```diff
@@ -0,0 +1,42 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Jake Potrebic <jake.m.potrebic@gmail.com>
+Date: Sat, 10 Dec 2022 17:52:38 -0800
+Subject: [PATCH] Deprecate and replace methods with old StructureType
+
+
+diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+index a05c1bc8874ef5e380544a6a344c848e37da49c4..4e8fde36a7a1b610a02efa839c2d099e0f167b33 100644
+--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
++++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+@@ -1844,6 +1844,31 @@ public final class CraftServer implements Server {
+ 
+         return CraftItemStack.asBukkitCopy(stack);
+     }
++    // Paper start - copied from above (uses un-deprecated StructureType type)
++    @Override
++    public ItemStack createExplorerMap(World world, Location location, org.bukkit.generator.structure.StructureType structureType, org.bukkit.map.MapCursor.Type mapIcon) {
++        return this.createExplorerMap(world, location, structureType, mapIcon, 100, true);
++    }
++
++    @Override
++    public ItemStack createExplorerMap(World world, Location location, org.bukkit.generator.structure.StructureType structureType, org.bukkit.map.MapCursor.Type mapIcon, int radius, boolean findUnexplored) {
++        Preconditions.checkArgument(world != null, "World cannot be null");
++        Preconditions.checkArgument(structureType != null, "StructureType cannot be null");
++        Preconditions.checkArgument(mapIcon != null, "mapIcon cannot be null");
++
++        ServerLevel worldServer = ((CraftWorld) world).getHandle();
++        Location structureLocation = world.locateNearestStructure(location, structureType, radius, findUnexplored).getLocation();
```

Comment:
This will throw a NPE if the structure is not found in the radius specified, that was already the case on the old methods. Maybe the method should be nullable?

Reply:
It did do this before... I will make this method nullable and change the deprecated one to throw an exception if it doesn't find a location instead of a random NPE.